File: golang/1967_num_of_strings/num_of_strings.go
// Package main ...
package main

import "fmt"

func numOfStrings(patterns []string, word string) int {
	check := func(pattern string, word string) bool {
		m := len(pattern)
		n := len(word)
		for i := 0; i+m <= n; i++ {
			flag := true
			for j := 0; j < m; j++ {
				if word[i+j] != pattern[j] {
					flag = false
					break
				}
			}
			if flag {
				return true
			}
		}
		return false
	}

	res := 0
	for _, pattern := range patterns {
		if check(pattern, word) {
			res++
		}
	}
	return res
}

func main() {
	tests := []struct {
		patterns []string
		word     string
		ans      int
	}{
		{[]string{"a", "abc", "bc", "d"}, "abc", 3},
		{[]string{"a", "b", "c"}, "aaaaabbbbb", 2},
		{[]string{"a", "a", "a"}, "ab", 3},
	}

	for _, test := range tests {
		res := numOfStrings(test.patterns, test.word)
		if res != test.ans {
			fmt.Println("test failed", test.patterns, test.word, res, test.ans)
		}
	}
}

File: cpp/1967_num_of_strings/num_of_strings.cpp
#include <cassert>
#include <string>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  int numOfStrings(vector<string>& patterns, string word) {
    auto check = [](const string& pattern, const string& word) -> bool {
      int m = pattern.size();
      int n = word.size();
      for (int i = 0; i + m <= n; ++i) {
        bool flag = true;
        for (int j = 0; j < m; ++j) {
          if (word[i + j] != pattern[j]) {
            flag = false;
            break;
          }
        }
        if (flag) {
          return true;
        }
      }
      return false;
    };

    int res = 0;
    for (const string& pattern : patterns) {
      res += check(pattern, word);
    }
    return res;
  }
};

int main() {
  vector<tuple<vector<string>, string, int>> tests{
      {{"a", "abc", "bc", "d"}, "abc", 3},
      {{"a", "b", "c"}, "aaaaabbbbb", 2},
      {{"a", "a", "a"}, "ab", 3},
  };

  for (auto& [patterns, word, ans] : tests) {
    assert(Solution().numOfStrings(patterns, word) == ans);
  }
}